### PR TITLE
Plp mvvm

### PR DESCRIPTION
**Pull request by sunnypatelbv**

### CODE REVIEW

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class PLPViewModel{
+    
+    //MARK: Variables
+    public weak var delegate: PLPViewModelDelegate?
+    var productArr = [Hit]()
+    var counter = 24
+    let url =  "https://ov-dev.sssports.com/s/UAE/dw/shop/v20_10/product_search?client_id=ce6abb4e-faf1-41af-94e7-feb1e2dd4a77&count=24&expand=prices,%20images,%20represented_products&locale=en-AE&refine=cgid%3Dbrands_nike&refine_1=htype%3Dvariation_group%7Cproduct&refine_2=price%3D(1..)&refine_3=orderable_only%3Dtrue&start=0"
+    
+    func getData(){
+        APIManager.shared.fetchData(pageUrl: url, dataModel: ProductInformationModel.self, completionHandler: {[weak self] (data) in
```
Comment (NayanB-cloud) [inline SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift]: How to check error from Api? as this completion will give only model.

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            self!.productArr = (data?.hits)!
```
Comment (NayanB-cloud) [inline SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift]: Why self is force caste?

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewController/ProductDetailCollectionCell/ProductDetailCollectionCell.swift [hunk 3]
```diff
@@ -100,7 +100,7 @@ class ProductDetailCollectionCell: UICollectionViewCell {
                 let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
                 print(json)
                 let wishlistData = try? JSONDecoder().decode(AddWishlistModel.self, from: data)
```
Comment (NayanB-cloud) [inline SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewController/ProductDetailCollectionCell/ProductDetailCollectionCell.swift]: Where is try catch for JsonDescoder?

**Review by NayanB-cloud — COMMENTED** [review]
Please check the comments.

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift (hunk 1 above)
Comment (sunnypatelbv) [reply SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift]: Changed the code, which will print the error in console, i checked the error printed in console by changing model

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift (hunk 2 above)
Comment (sunnypatelbv) [reply SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewModel/PLPViewModel.swift]: Changed to optional

File: SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewController/ProductDetailCollectionCell/ProductDetailCollectionCell.swift (hunk 3 above)
Comment (sunnypatelbv) [reply SSS_task/ViewControllers/CategoryControllers/PLP_VC/ViewController/ProductDetailCollectionCell/ProductDetailCollectionCell.swift]: removed it, used it only to check if data is receiving or not

**Comment by NayanB-cloud** [comment]
Yes changes are done.

**Pull request merged**